src/crimson/osd/pg_recovery.cc

```diff
@@ -259,8 +259,8 @@ size_t PGRecovery::start_backfill_ops(
   size_t max_to_start,
   std::vector<crimson::osd::blocking_future<>> *out)
 {
-  assert(!pg->get_peering_state().get_backfill_targets().empty());
-
+  if (pg->get_peering_state().get_backfill_targets().empty())
```

Comment:
better off putting the rationale in the commit message, IIUC,  `osd_recovery_max_single_start` is not necessarily used up when `start_backfill_ops()` gets called even if it's the last consumer of this setting.

Reply:
@xuxuehan could you reconsider this comment?

Reply:
> better off putting the rationale in the commit message, IIUC, `osd_recovery_max_single_start` is not necessarily used up when `start_backfill_ops()` gets called even if it's the last consumer of this setting.

Yes, that's right, I'll put it in the commit message:-)